ospfd/ospf_ext.c

```diff
@@ -264,18 +286,43 @@ static struct ext_itf *lookup_ext_by_ifp(struct interface *ifp)
 	return NULL;
 }
 
+/* Lookup prefix sid LSA by LSA ID from the extended prefix struct */
+static struct prefix_sid_lsa *
+lookup_psid_lsa_by_instance(const struct ext_itf *exti, const uint32_t instance)
+{
+	struct listnode *node;
+	struct prefix_sid_lsa *psid;
+
+	if (exti->type != OPAQUE_TYPE_EXTENDED_PREFIX_LSA)
+		return NULL;
+
+	for (ALL_LIST_ELEMENTS_RO(exti->lsa.prefix_sid_list, node, psid))
+		if (psid->instance == instance)
+			return psid;
+
+	return NULL;
+}
+
 /* Lookup Extended Prefix/Links by LSA ID from OspfEXT struct iflist */
 static struct ext_itf *lookup_ext_by_instance(struct ospf_lsa *lsa)
 {
 	struct listnode *node;
 	struct ext_itf *exti;
+	struct prefix_sid_lsa *psid;
 	uint32_t key = GET_OPAQUE_ID(ntohl(lsa->data->id.s_addr));
 	uint8_t type = GET_OPAQUE_TYPE(ntohl(lsa->data->id.s_addr));
 
 
-	for (ALL_LIST_ELEMENTS_RO(OspfEXT.iflist, node, exti))
-		if ((exti->instance == key) && (exti->type == type))
+	for (ALL_LIST_ELEMENTS_RO(OspfEXT.iflist, node, exti)) {
+		/* if it's an extended prefix LSA, check all of them */
+		if (exti->type == OPAQUE_TYPE_EXTENDED_PREFIX_LSA) {
+			psid = lookup_psid_lsa_by_instance(exti, key);
+			if (psid)
+				return exti;
+		} else if ((exti->lsa.link_lsa.instance == key)
+			   && (exti->type == type))
```

Comment:
Same as above. Perhaps an alternative is to move instance field from the prefix SID list